tests/ops/integration_tests/test_sql_task.py

```diff
@@ -1089,3 +1094,315 @@ async def test_retry_erasure(
             ("postgres_example_test_dataset:address", "error"),
         ]
+
+
+@pytest.mark.integration_timescale
+@pytest.mark.integration
+@pytest.mark.asyncio
+async def test_timescale_access_request_task(
+    db,
+    policy,
+    timescale_connection_config,
+    timescale_integration_db,
+) -> None:
+    database_name = "my_timescale_db_1"
+    privacy_request = PrivacyRequest(
+        id=f"test_timescale_access_request_task_{random.randint(0, 1000000)}"
+    )
+
+    v = await graph_task.run_access_request(
+        privacy_request,
+        policy,
+        integration_db_graph(database_name),
+        [timescale_connection_config],
+        {"email": "customer-1@example.com"},
+        db,
+    )
+
+    assert_rows_match(
+        v[f"{database_name}:address"],
+        min_size=2,
+        keys=["id", "street", "city", "state", "zip"],
+    )
+    assert_rows_match(
+        v[f"{database_name}:orders"],
+        min_size=3,
+        keys=["id", "customer_id", "shipping_address_id", "payment_card_id"],
+    )
+    assert_rows_match(
+        v[f"{database_name}:payment_card"],
+        min_size=2,
+        keys=["id", "name", "ccn", "customer_id", "billing_address_id"],
+    )
+    assert_rows_match(
+        v[f"{database_name}:customer"],
+        min_size=1,
+        keys=["id", "name", "email", "address_id"],
+    )
+
+    # links
+    assert v[f"{database_name}:customer"][0]["email"] == "customer-1@example.com"
+
+    logs = (
+        ExecutionLog.query(db=db)
+        .filter(ExecutionLog.privacy_request_id == privacy_request.id)
+        .all()
+    )
+
+    logs = [log.__dict__ for log in logs]
+
+    assert (
+        len(
+            records_matching_fields(
+                logs, dataset_name=database_name, collection_name="customer"
+            )
+        )
+        > 0
+    )
+
+    assert (
+        len(
+            records_matching_fields(
+                logs, dataset_name=database_name, collection_name="address"
+            )
+        )
+        > 0
+    )
+
+    assert (
+        len(
+            records_matching_fields(
+                logs, dataset_name=database_name, collection_name="orders"
+            )
+        )
+        > 0
+    )
+
+    assert (
+        len(
+            records_matching_fields(
+                logs,
+                dataset_name=database_name,
+                collection_name="payment_card",
+            )
+        )
+        > 0
+    )
+
+
+@pytest.mark.integration_timescale
+@pytest.mark.integration
+@pytest.mark.asyncio
+async def test_timescale_erasure_request_task(
+    db,
+    erasure_policy,
+    timescale_connection_config,
+    timescale_integration_db,
+) -> None:
+    rule = erasure_policy.rules[0]
+    target = rule.targets[0]
+    target.data_category = "user"
+    target.save(db)
+
+    database_name = "my_timescale_db_1"
+    privacy_request = PrivacyRequest(
+        id=f"test_timescale_access_request_task_{random.randint(0, 1000000)}"
```

Comment:
Is the random number important? It would be simpler and more reproducible without it, but if it is needed it's ok.

Is it to keep it from clashing with the other tests? If so there is the chance the numbers match causing random failures (small chance but still a chance).

Reply:
We might not need the random number specifically, but the main thing we want is for this id not to clash with other privacy request ids in the redis cache because that will affect request execution which relies heavily on the cache.

The privacy request fixtures that we use elsewhere are recreated for each test so their id is different. Here, we're not using a fixture, but creating a resource that isn't persisted to the db (but its data is persisted to the cache).

This is a pattern that we follow elsewhere in the tests on this page - I increased from 1000 in other tests to 1000000. Open to more ideas around this because I do think occasional collisions are part of our testing issues.

Reply:
Is it required that it is a number? Would a uuid work?

Reply:
Yes I could change it to a uuid thanks